src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -91,16 +100,21 @@
     LessThanOrEqualToSegment=KeywordSegment.make('<=', name='less_than_equal_to', type='comparison_operator'),
     NotEqualToSegment_a=KeywordSegment.make('!=', name='not_equal_to', type='comparison_operator'),
     NotEqualToSegment_b=KeywordSegment.make('<>', name='not_equal_to', type='comparison_operator'),
+    # Keywords # TODO Check if ANSI compliant:
+    NanKeywordSegment=KeywordSegment.make('nan'),
+    OverwriteKeywordSegment=KeywordSegment.make('overwrite'),
+    ValueKeywordSegment=KeywordSegment.make('value'),
     # The strange regex here it to make sure we don't accidentally match numeric literals. We
     # also use a regex to explicitly exclude disallowed keywords.
     NakedIdentifierSegment=ReSegment.make(
         r"[A-Z0-9_]*[A-Z][A-Z0-9_]*", name='identifier', type='naked_identifier',
-        _anti_template=r"^(SELECT|JOIN|ON|USING|CROSS|INNER|LEFT|RIGHT|OUTER|INTERVAL|CASE|FULL|NULL)$"),
+        _anti_template=r"^(" + '|'.join(anti_template) + ")$"),
     FunctionNameSegment=ReSegment.make(r"[A-Z][A-Z0-9_]*", name='function_name', type='function_name'),
     # Maybe data types should be more restrictive?
     DatatypeIdentifierSegment=ReSegment.make(r"[A-Z][A-Z0-9_]*", name='data_type_identifier', type='data_type_identifier'),
-    # Maybe date parts should be more restrictive
-    DatepartSegment=ReSegment.make(r"[A-Z][A-Z0-9_]*", name='date_part', type='date_part'),
+    # Ansi Intervals
+    IntervalSegment=ReSegment.make(r"^(DAY|DAYOFYEAR|HOUR|MILLISECOND|MINUTE|MONTH|QUARTER|SECOND|WEEK|WEEKDAY|YEAR)$",
```

Comment:
I think it should still be called `DatePartSegment`, because the same segment could also be used in other contexts outside of an `Interval`, for example inside a function call.

Reply:
Date, Time and Interval are all reserved SQL words. A DATE field consists of the year, month, and day. TIME contains hour, minute, and second values. An INTERVAL consists of an INTERVAL a single datetime field, e.g. year, month, day, hour, minute, second, (millisecond not in SQL 1992 standard). Thus my motivation to change it, since it is not a subset of the date fields.
The defined object is used at two positions in the code of dialect_ansi.py. In both cases it refers to a single datetime field. How about I clean up the definition (only year, month, day, hour, minute, second, millisecond) and we call it DatetimeField?

Reply:
I've done a quick review of some of the dialects that use this:
- mysql call it a `unit`: https://dev.mysql.com/doc/refman/8.0/en/expressions.html#temporal-intervals
- snowflake call it a date or time `part`: https://dev.mysql.com/doc/refman/8.0/en/expressions.html#temporal-intervals
- teradata call it the `qualifier`: https://docs.teradata.com/reader/kmuOwjp1zEYg98JsB8fu_A/Njb3o4V_Gi0K7FJPN2KAJQ

It looks like there isn't much consensus in what this bit is called. Given the oldest of those dialects in mysql, I'm tempted to default to that one and call it `DatetimeUnitSegment` or some close derivative of that.

Reply:
Thanks a lot for checking. DateTimeUnitSegment sounds good to me.